src/organizercore.cpp

```diff
@@ -1891,15 +1889,12 @@ bool OrganizerCore::saveCurrentLists()
 
 void OrganizerCore::savePluginList()
 {
-  if (m_DirectoryUpdate) {
-    // delay save till after directory update
-    m_PostRefreshTasks.append([this]() {
-      this->savePluginList();
-    });
-    return;
-  }
-  m_PluginList.saveTo(m_CurrentProfile->getLockedOrderFileName());
-  m_PluginList.saveLoadOrder(*m_DirectoryStructure);
+  onNextRefresh(
+      [this]() {
+        m_PluginList.saveTo(m_CurrentProfile->getLockedOrderFileName());
+        m_PluginList.saveLoadOrder(*m_DirectoryStructure);
+      },
+      RefreshCallbackGroup::CORE);
```

Comment:
Same here, doesn't this mean it might not save the list until a future refresh is triggered?
Shouldn't it check if a refresh is happening and only then register the onNextRefresh?

Reply:
As above, this makes it look like the function will only be called once the next refresh is started, rather than as soon as there are no refreshes ongoing.